src/server/scripts/Northrend/Naxxramas/boss_thaddius.cpp

```diff
@@ -110,168 +154,302 @@ class boss_thaddius : public CreatureScript
-                me->SetReactState(REACT_PASSIVE);
+                return (uint32)(id == DATA_POLARITY_CROSSED && shockingEligibility);
             }
-        }
-
-        void EnterCombat(Unit* /*who*/) override
-        {
-            _EnterCombat();
-            Talk(SAY_AGGRO);
-            events.ScheduleEvent(EVENT_SHIFT, 30000);
-            events.ScheduleEvent(EVENT_CHAIN, urand(10000, 20000));
-            events.ScheduleEvent(EVENT_BERSERK, 360000);
-        }
-
-        void DamageTaken(Unit* /*pDoneBy*/, uint32 & /*uiDamage*/) override
-        {
-            me->SetReactState(REACT_AGGRESSIVE);
-        }
 
-        void SetData(uint32 id, uint32 data) override
-        {
-            if (id == DATA_POLARITY_SWITCH)
-                polaritySwitch = data ? true : false;
-        }
+            void Transition() // initiate transition between pet phase and thaddius phase
+            {
+                _phase = PHASE_TRANSITION;
+                events.SetPhase(PHASE_TRANSITION);
 
-        uint32 GetData(uint32 id) const override
-        {
-            if (id != DATA_POLARITY_SWITCH)
-                return 0;
+                me->RemoveFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NOT_SELECTABLE);
 
-            return uint32(polaritySwitch);
-        }
+                events.ScheduleEvent(EVENT_TRANSITION_1, 10 * IN_MILLISECONDS, 0, PHASE_TRANSITION);
+                events.ScheduleEvent(EVENT_TRANSITION_2, 12 * IN_MILLISECONDS, 0, PHASE_TRANSITION);
+                events.ScheduleEvent(EVENT_TRANSITION_3, 15 * IN_MILLISECONDS, 0, PHASE_TRANSITION);
+            }
 
-        void UpdateAI(uint32 diff) override
-        {
-            if (checkFeugenAlive && checkStalaggAlive)
-                uiAddsTimer = 0;
+            // @todo remove debug
+            // debug: point to begin reset, laugh to force (improper) reset
+            /*void ReceiveEmote(Player* player, uint32 emoteId) override
+            {
+                if (emoteId == 72)
+                    BeginResetEncounter();
+                else if (emoteId == 60)
+                    ResetEncounter();
+            }*/
 
-            if (checkStalaggAlive != checkFeugenAlive)
+            void BeginResetEncounter()
             {
-                uiAddsTimer += diff;
-                if (uiAddsTimer > 5000)
-                {
-                    if (!checkStalaggAlive)
-                    {
-                        if (Creature* pStalagg = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_STALAGG)))
-                            pStalagg->Respawn();
-                    }
-                    else
-                    {
-                        if (Creature* pFeugen = ObjectAccessor::GetCreature(*me, instance->GetGuidData(DATA_FEUGEN)))
-                            pFeugen->Respawn();
-                    }
+                if (!me->IsAlive()) return;
+
+                if (_phase != PHASE_RESETTING)
+                    if(instance)
```

Comment:
Checking for instance is not required, script should only run in instanced maps.